File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    
    bool isPoss(vector<int> &arr, int x,int h){
        long long int total=0;
        for(int i=0;i<arr.size();i++){
            total= total + ceil((double)arr[i]/ (double)x);
        }
        return total<=h;
    }
    
    int minEatingSpeed(vector<int>& piles, int h) {
        int low = 1;
        int high = *max_element(piles.begin(), piles.end());
        
        while(low<=high){
            int mid = (low+high)/2;
            if(isPoss(piles, mid, h)){
                high = mid-1;
            }else{
                low = mid+1;
            }
        }
        return low;
    }
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.java
class Solution {
        
     public boolean isEat(int[] piles, int h, int x){
             long total = 0;
             for(int i=0; i<piles.length; i++){
                     total += Math.ceil((double)piles[i] / (double)x);
             }
             return total<=h;
     }   
        
    public int minEatingSpeed(int[] piles, int h) {
            int mini = 1;
            int maxi = Integer.MIN_VALUE;
            
            for(int i=0; i<piles.length; i++){
                    mini = Math.min(piles[i], mini);
                    maxi = Math.max(piles[i], maxi);
            }
            
            while(mini<=maxi){
                    int mid = (mini+maxi)/2;
                    
                    if(isEat(piles, h, mid)){
                            maxi = mid-1;
                    }else{
                            mini = mid+1;
                    }
                    
            }
            return mini;
    }
}